### PR TITLE
ci: nightly collect runtimes jobs results

### PR DESCRIPTION
**Pull request by nirLumigo**
Scenarios:
image

image
(this can also result in IndexOutOfRange)

image

image

image

- Open PR, example:
image

- session.install fix deprecated warning

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -1,71 +1,15 @@
...
-from dataclasses import dataclass
 from xml.etree import ElementTree
+import time
+from typing import List, Union, Dict
 
 import nox
 import requests
 from packaging.version import parse as parse_version, Version
 
-import os
```
Comment (nirLumigo) [inline noxfile.py:11]: Moved it to another file

File: .github/workflows/nightly-actions.yml [hunk 2]
```diff
@@ -1,27 +1,59 @@
...
-    name: check-new-verisons-of-instrumented-packages
+    name: check-new-versions-of-instrumented-packages
     steps:
       - uses: actions/checkout@v3
       - name: Set up Python
         uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
-          architecture: x64
-      - run: sudo apt update && sudo apt install -y libxml-xpath-perl git
```
Comment (nirLumigo) [inline .github/workflows/nightly-actions.yml:19]: I think that `xpath` isn't needed anymore

File: .github/workflows/nightly-actions.yml [hunk 3]
```diff
@@ -1,27 +1,59 @@
 name: Version Testing
-on:
-  schedule:
-    - cron: "0 0 * * *"
+on: [push]
```
Comment (nirLumigo) [inline .github/workflows/nightly-actions.yml:2]: I need to revert it before merging

File: src/ci/tested_versions_utils.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
+from __future__ import annotations
```
Comment (nirLumigo) [inline src/ci/tested_versions_utils.py:1]: Most of this file is copied, I'll mark the changed parts

File: src/ci/tested_versions_utils.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        TestedVersions.add_version_to_file(
+            dependency_file_path, dependency_version, success
+        )
+
+    @staticmethod
+    def add_version_to_file(path: str, version: str, success: bool):
+        new_line = f"{'' if success else '!'}{version}\n"
+        print(f"Adding the following line to {path}: {new_line}")
+        with open(path, "a") as f:
+            f.write(new_line)
```
Comment (nirLumigo) [inline src/ci/tested_versions_utils.py:28]: It was one method and I splitted it to 2.

File: src/ci/tested_versions_utils.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+            for line in f:
+                line = line.strip()
+                if line.startswith("!"):
+                    failed.append(line[1:])
+                else:
+                    success.append(line)
+        return TestedVersions(success=tuple(success), failed=tuple(failed))
+
+    def get_all_versions(self) -> Tuple[str, ...]:
+        return self.success + self.failed
```
Comment (nirLumigo) [inline src/ci/tested_versions_utils.py:70]: Added this method

File: scripts/push_versions_branch.sh [hunk 7]
```diff
@@ -1,5 +1,6 @@
 if [[ ! `git status --porcelain --untracked-files=no` ]]; then echo "no changes, skipping commit" && exit 0; fi
 
+git config user.email "versionbot@lumigo.io" && git config user.name "Version Bot"
```
Comment (nirLumigo) [inline scripts/push_versions_branch.sh:3]: Moved it here from nightly-actions.yml

**Review by GuyMoses — APPROVED** [review]

**Pull request merged**